extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/processorsupplier/TaskMaxProcessorMetaSupplier.java

```diff
@@ -0,0 +1,133 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+package com.hazelcast.jet.kafka.connect.impl.processorsupplier;
+
+import com.hazelcast.cluster.Address;
+import com.hazelcast.jet.core.ProcessorMetaSupplier;
+import com.hazelcast.jet.core.ProcessorSupplier;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.function.Function;
+
+/**
+ * This class distributes specified number of processors evenly among cluster members
+ */
+public class TaskMaxProcessorMetaSupplier implements ProcessorMetaSupplier, DataSerializable {
+    private int tasksMax;
+    private ReadKafkaConnectProcessorSupplier supplier;
+    private boolean partitionedAddresses;
+    private final List<Address> memberAddressList = new ArrayList<>();
+    private final List<Integer> memberLocalParallelismList = new ArrayList<>();
+    private int processorOrder;
+
+    public void setTasksMax(int tasksMax) {
+        this.tasksMax = tasksMax;
+    }
+
+    public void setSupplier(ReadKafkaConnectProcessorSupplier supplier) {
+        this.supplier = supplier;
+    }
+
+    @Override
+    public void init(@Nonnull Context context) {
+        int totalParallelism = context.totalParallelism();
+        if (totalParallelism < tasksMax) {
+            throw new IllegalArgumentException("The requested parallelism of " + tasksMax + " is greater than " +
+                                               "the available parallelism of " + totalParallelism +
+                                               " for Kafka Connect vertices. " +
+                                               "Please call setLocalParallelism(" + tasksMax + ") " +
+                                               "for Kafka Connect Source"
+            );
+        }
+    }
+
+    @Nonnull
+    @Override
+    public Function<? super Address, ? extends ProcessorSupplier> get(@Nonnull List<Address> addresses) {
+        if (!partitionedAddresses) {
+            partitionedAddresses = true;
+            partitionTasks(addresses);
+        }
+        return memberAddress -> {
```

Comment:
I would personally pack exactly `localParallelism` tasks per PS as it's the standard behaviour. Now we basically ignore what user sets in the pipeline

Reply:
The original design was to distribute tasks.max runners among members evenly. 
To satisfy local parallelism constraint ExpectNothingProcessor was used. 

With your changes, local parallelism count of ReadKafkaConnect processors are created, but some of these processors may not be active

So now we are not taking into account tasks.max upper limit.